# File containing task definitions.

# INITIALIZE.
initialize:
  actions:
    - initialize:
        countdown: 10
        drift_check: 3

# GATE TASK
# The gate is a purely dead reckoning task.
gate:
  actions:
    - sleep: ""
    - move_all:
        yaw: 0.0
        depth: 1.4
        distance: 25.0
    - move_all:
        yaw: -0.55
        depth: 2.35
        distance: 20.0
    - move_all:
        depth: 1.4
      #- roulette:
      #  depth: 1.4
      # - move_all:
      #  depth: 3.0
      #  distance: -0.1
    - roll:
        depth: 3.0
        angle: 2.0
    - move_all:
        depth: 2.0
        distance: 25.0
# BUOY TASK
buoy:
  actions: []

# MANEUVER TASK
# Maneuver with feedback. Might want to do something fancy at the end for
# style points!
maneuver:
  actions: []

# BINS TASK
# Drop marker in bins.
bins:
  actions:
    #- acoustic_servo:
    #    depth: 2.0
    - roulette:
        depth: 3.0
    - roll:
        depth: 3.0
        angle: 2.0
    - move_all:
        depth: 2.5
        distance: 5.0
# TORPEDO TASK
# Move to torpedos and then shoot.
torpedo:
  actions: []

# OCTAGON TASK
# Search for octogon using hydrophones.
octagon:
  actions:
    - move_all:
        depth: 0.40
    - roll:
        depth: 0.4
        angle: 2.0

#TEST TASK]


# Represents any series of actions for testing purposes.
square:
  actions:
  #go down
    - move:
        depth: 1.0
        distance: 5.0
        heading: 0.76 # on [-pi, pi]

#FOLLOW LANE
#Finds a Lane, servos to center and turns
follow_lane:
  actions:
    - move_all:
        depth: 2.0
    - follow_lane:
        depth: 2.0
    - move_all:
        depth: 2.0
        distance: 10.0
#ROULETTE TASK
#Looks for roulette and servos to the center
roulette:
  actions:
    - roulette: "not needed"

dice:
  actions:
    - dice_servo:
        diceName: 'dice6'
    - move:
        distance: 1.5
    - move:
        distance: -1.5
    - dice_servo:
        diceName: 'dice5'
    - move:
        distance: 1.5

